backend/app/src/main/java/com/moh/phlat/backend/repository/ProcessDataRepository.java

```diff
@@ -33,4 +35,413 @@ public interface ProcessDataRepository extends JpaRepository<ProcessData, Long>
+	@Query(value="Select distinct PHYSICAL_ADDR4 FROM process_data WHERE control_id = :controlTableId",nativeQuery=true)
+	public List<String> findAllDistinctPhysicalAddr4(Long controlTableId);
+	@Query(value="Select distinct PHYSICAL_CITY FROM process_data WHERE control_id = :controlTableId",nativeQuery=true)
+	public List<String> findAllDistinctPhysicalCity(Long controlTableId);
+	@Query(value="Select distinct PHYSICAL_PROVINCE FROM process_data WHERE control_id = :controlTableId",nativeQuery=true)
+	public List<String> findAllDistinctPhysicalProvince(Long controlTableId);
+	@Query(value="Select distinct PHYSICAL_PCODE FROM process_data WHERE control_id = :controlTableId",nativeQuery=true)
+	public List<String> findAllDistinctPhysicalPcode(Long controlTableId);
+	@Query(value="Select distinct PHYSICAL_COUNTRY FROM process_data WHERE control_id = :controlTableId",nativeQuery=true)
+	public List<String> findAllDistinctPhysicalCountry(Long controlTableId);
+	@Query(value="Select distinct PHYS_ADDR_IS_PRIVATE FROM process_data WHERE control_id = :controlTableId",nativeQuery=true)
+	public List<String> findAllDistinctPhysAddrIsPrivate(Long controlTableId);
+	@Query(value="Select distinct MAIL_ADDR1 FROM process_data WHERE control_id = :controlTableId",nativeQuery=true)
+	public List<String> findAllDistinctMailAddr1(Long controlTableId);
+	@Query(value="Select distinct MAIL_ADDR2 FROM process_data WHERE control_id = :controlTableId",nativeQuery=true)
+	public List<String> findAllDistinctMailAddr2(Long controlTableId);
+	@Query(value="Select distinct MAIL_ADDR3 FROM process_data WHERE control_id = :controlTableId",nativeQuery=true)
+	public List<String> findAllDistinctMailAddr3(Long controlTableId);
+	@Query(value="Select distinct MAIL_ADDR4 FROM process_data WHERE control_id = :controlTableId",nativeQuery=true)
+	public List<String> findAllDistinctMailAddr4(Long controlTableId);
+	@Query(value="Select distinct MAIL_CITY FROM process_data WHERE control_id = :controlTableId",nativeQuery=true)
+	public List<String> findAllDistinctMailCity(Long controlTableId);
+	@Query(value="Select distinct MAIL_BC FROM process_data WHERE control_id = :controlTableId",nativeQuery=true)
+	public List<String> findAllDistinctMailBc(Long controlTableId);
+	@Query(value="Select distinct MAIL_PCODE FROM process_data WHERE control_id = :controlTableId",nativeQuery=true)
+	public List<String> findAllDistinctMailPcode(Long controlTableId);
+	@Query(value="Select distinct MAIL_COUNTRY FROM process_data WHERE control_id = :controlTableId",nativeQuery=true)
+	public List<String> findAllDistinctMailCountry(Long controlTableId);
+	@Query(value="Select distinct MAIL_ADDR_IS_PRIVATE FROM process_data WHERE control_id = :controlTableId",nativeQuery=true)
+	public List<String> findAllDistinctMailAddrIsPrivate(Long controlTableId);
+	
+	public static Specification<ProcessData> buildSpecificationIn(Long controlId, String reqRowStatusCode, List<String> ids, List<String> actions, 
```

Comment:
specification should be in separate class of it's own

Reply:
Will separate specification.